CHANGELOG.md

```diff
@@ -2,6 +2,24 @@
 
 All notable changes to the Zowe API Mediation Layer package will be documented in this file.
 
+## `2.17.0 (2024-07-12)`
```

Comment:
The previous release has the specific API ML Released, so this should probably be 2.17.2

Also, there are two entries in the changelog for 2.16 (2.16.0 and 2.16.2) I know it's not part of this PR but it needs to be fixed. Also add the Zowe version that it belongs to (like in 2.16.0 entry)